src/frontend/js/components/PaymentButton/components/PaymentInterfaces/PayplugLightbox.tsx

```diff
@@ -1,21 +1,23 @@
 import { useCallback, useEffect, useRef } from 'react';
 import { handle } from 'utils/errors/handle';
-import { OneClickPaymentInfo, PaymentErrorMessageId } from 'components/PaymentButton';
-import { Payment } from 'types/Joanie';
-import type { PaymentInterfaceProps } from '.';
+import { PaymentErrorMessageId } from 'components/PaymentButton';
+import { PayplugPayment, PaymentInterfaceProps } from 'types/Joanie';
 
 /**
  * Load the form payplug script if it has been not yet uploaded then open
  * the lightbox and listen messages coming from it.
  *
  * https://docs.payplug.com/api/lightbox.html#lightbox
  */
-const PayplugLightbox = ({ onSuccess, onError, ...props }: PaymentInterfaceProps) => {
+const PayplugLightbox = ({
+  onSuccess,
+  onError,
+  ...props
+}: PaymentInterfaceProps<PayplugPayment>) => {
   const ref = useRef<ReturnType<typeof setTimeout>>();
 
   /** type guard to check if the payment is a payment one click */
-  const isOneClickPayment = (p: Payment): p is OneClickPaymentInfo =>
-    (p as OneClickPaymentInfo)?.is_paid === true;
+  const isOneClickPayment = (p: PayplugPayment) => p?.is_paid === true;
```

Comment:
the correct typeguard would be 
```ts
    const isOneClickPayment = (p: PayplugPayment) => p?.is_paid !== undefined;
```

Reply:
If `is_paid` is false, the idea is to consider the payment as a regular one (open the payment interface then call onSuccess callback when needed). If is_paid is true, the onSuccess callback is called immediately without invoking the PaymentInterface.